### DISCUSSION TITLE
Package leap frog

### DISCUSSION THREAD

**Discussion by KirkPen** [microsoft/vcpkg#29683] [Q&A] [answered]
I might be doing something wrong, but, when I install vcpkg install opencv[cuda,tbb] then find out I also need pcl[cuda] and installing the pcl[cuda] re installs the opencv and that takes 4 hours, then i need another and it reinstalls something else, what is the answer to this? I don't know what package needs what other package. 

And I don't know what order to install them so the don't just keep reinstalling each other.
Any help?

**Comment by dg0yt** [comment, marked as answer]
The install order is defined by explicit dependencies between ports and features. You don't have to know the order. 
However, you need to know which features you want from the ultimate packages, and sometimes also from selected dependencies.
Then you can pass them all at once, e.g. `vcpkg install opencv[cuda,tbb] pcl[cuda]`.

To get a picture of dependencies, use `vcpkg depend-info opencv[cuda,tbb]`.

Sometimes it is also useful to run `vcpkg install --dry-run opencv[cuda,tbb]`. This won't list any packages which are already installed with a sufficient feature set. But it can be used with other triplets - I regularly use this from linux for windows and osx triplets. Unlike `depend-info`, it also lists features.

If installation order changes build results, it should be reported as a bug.

**Reply by KirkPen** [reply]
Ok thank you.
